snuba/datasets/entities/transactions.py

```diff
@@ -86,20 +99,125 @@
 )
 
 
+class SampledSimplePipelineBuilder(SimplePipelineBuilder):
+    def build_planner(
+        self, query: LogicalQuery, settings: RequestSettings,
+    ) -> EntityQueryPlanner:
+        new_query = deepcopy(query)
+        sampling_rate = state.get_config("snuplicator-sampling-rate", 1.0)
+        assert isinstance(sampling_rate, float)
+        new_query.set_sample(sampling_rate)
+        return super().build_planner(new_query, settings)
+
+
+def sampling_selector_func(query: LogicalQuery, referrer: str) -> Tuple[str, List[str]]:
+    if referrer == "tagstore.__get_tag_keys":
+        condition = query.get_condition()
+        assert condition is not None
+        project_ids = get_project_ids_in_query_ast(query, "project_id")
+        if not project_ids:
+            return "transactions", []
+
+        test_projects_raw = state.get_config("snuplicator-sampling-projects", "")
+        test_projects = set()
+        if isinstance(test_projects_raw, str):
+            test_projects = set(int(p) for p in test_projects_raw.split(",") if p)
```

Comment:
`if p != ""`
PEP8 suggest to distinguish between the cases where you want None and those where you want a falsey value.
More important, I think, in this case, you would make it clear which corner case you would be covering with the if that may not be obvious